### PR TITLE
Clean Up Block Retrieval Methods

### PR DESCRIPTION
**Pull request by nisdas**
**What type of PR is this?**

Code Quality

**What does this PR do? Why is it needed?**

 - [x] Cleans up block retrieval methods.
-  [x] Add in a few additional test cases for block retrieval.
 
**Which issues(s) does this PR fix?**

Fixes #

**Other notes for review**

### CODE REVIEW

File: beacon-chain/db/kv/blocks.go [hunk 1]
```diff
@@ -415,20 +418,18 @@ func fetchBlockRootsBySlotRange(
...
+		return key != nil && bytes.Compare(key, max) <= 0
 	}
-	rootsRange := (endSlot - startSlot) / step
 	if endSlot < startSlot {
-		rootsRange = 0
+		return nil, errInvalidSlotRange
+	}
+	// Return nothing with an end slot of 0.
+	if endSlot == 0 {
+		return [][]byte{}, nil
```
Comment (rkapka) [inline beacon-chain/db/kv/blocks.go:430]: Shouldn't this return the genesis block?

**Review by rauljordan — APPROVED** [review]

File: beacon-chain/db/kv/blocks.go (hunk 1 above)
Comment (rauljordan) [reply beacon-chain/db/kv/blocks.go:430]: We always retrieve the genesis block by its root and not by slots to overcome the limitations of zero values in Go throughout the codebase

Comment (rauljordan) [reply beacon-chain/db/kv/blocks.go:430]: Perhaps worth adding as a comment @nisdas

**Review by rauljordan — APPROVED** [review]

**Comment by rauljordan** [comment]
@nisdas looks like some DB tests fail

File: beacon-chain/state/stategen/replay.go [hunk 2]
```diff
@@ -53,6 +53,10 @@ func (s *State) ReplayBlocks(ctx context.Context, state *stateTrie.BeaconState,
 // LoadBlocks loads the blocks between start slot and end slot by recursively fetching from end block root.
 // The Blocks are returned in slot-descending order.
 func (s *State) LoadBlocks(ctx context.Context, startSlot, endSlot uint64, endBlockRoot [32]byte) ([]*ethpb.SignedBeaconBlock, error) {
+	// Nothing to load for invalid range.
+	if endSlot < startSlot {
+		return nil, nil
```
Comment (prestonvanloon) [inline beacon-chain/state/stategen/replay.go:58]: Return error?

Comment (prestonvanloon) [reply beacon-chain/state/stategen/replay.go:59]: Discussed offline that this breaks many tests. We'll revisit at a later time. @nisdas suggested filing an issue and TODO for this, to fix the test input data.

**Review by prestonvanloon — APPROVED** [review]

**Pull request merged**